### ISSUE TITLE
FB13552833: `SetFocusFilterIntent#perform` is only called when a focus is activated, but not when focus is deactivated

### ISSUE THREAD

**Issue by sindresorhus** [feedback-assistant/reports#461]
- **Date:** 2024-01-20
- **Resolution:** Open
- **Area:** App Intents
- **OS:** macOS 14.2
- **Type:** Incorrect/Unexpected Behavior

## Description

I tried to implement a focus filter for my calendar app, and I noticed that the `perform` method is only called when the user activates a focus like “Do Not Disturb”, but not when they deactivate the focus. I use `perform` to set some filter state in my app, and since `perform` is not called when the user deactives the focus, I’m not able to reset the filter state. That means the user will still see a filtered state in my app, even though they have deactivated the focus.

```swift
struct FilterCalendarsFocusFilter: SetFocusFilterIntent {
	static let title: LocalizedStringResource = "Filter Calendars"

	static let description = IntentDescription("")

	var displayRepresentation: DisplayRepresentation = ""

	@Parameter(
		title: "Calendars",
		optionsProvider: CalendarsOptionsProvider()
	)
	var calendars: [String]

	func perform() async throws -> some IntentResult {
		// Only called when the user activates "Do Not Disturb", but not when they deactivate.
		print("FOCUS FILTER PERFORM")
		return .result()
	}
}
```

tl;dr `SetFocusFilterIntent#perform` should be called when the user deactives a focus too.

I am aware there is `FilterCalendarsFocusFilter.current`, which I could poll on an interval, but that’s very inefficient.
